### PR TITLE
Interactions overhaul

### PR DESCRIPTION
**Pull request by stonier**
Mostly about getting interactions free from the concert.
- `rocon_python_comms` gets previous comms related functions from `rocon_utilities`
- `rocon_python_comms` gets `find_service` and `find_topic`.
- `rocon_python_utils` gets all network, system and ros functions from `rocon_utilities`
- `rocon_python_wifi` from the `rocon_multimaster` repo
- `rocon_python_redis` from the `rocon_multimaster` repo
- `rocon_semantic_version` is a nice module for playing with version numbers
- `rocon_master_info` is a de-concertified version of the olde `concert_info`
- `rocon_interactions` freed from the concert and finds a home here.

Brought to you by https://github.com/robotics-in-concert/rocon_concert/issues/188

@jihoonl, @bit-pirate can you take a quick gander over what's going in here to keep abreast of changes.

### CODE REVIEW

**Comment by jihoonl** [comment]
@bit-pirate ?

**Comment by bit-pirate** [comment]
looking into it

File: rocon_interactions/src/rocon_interactions/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#
+# License: BSD
+#   https://raw.github.com/robotics-in-concert/rocon_concert/license/LICENSE
+#
+##############################################################################
+# Exceptions
+##############################################################################
+
```
Comment (bit-pirate) [inline rocon_interactions/src/rocon_interactions/exceptions.py:8]: Is there no import required?

File: rocon_master_info/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+##############################################################################
+
+install(
+    PROGRAMS 
+        scripts/rocon_master_info
+        scripts/master.py
+    DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+    )
+
+#install(DIRECTORY launch
```
Comment (bit-pirate) [inline rocon_master_info/CMakeLists.txt]: remove?

File: rocon_python_redis/package.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  <name>rocon_python_redis</name>
+  <version>0.1.0</version>
+  <description>
+    Locally patched version of the python redis client.  </description>
+  <maintainer email="d.stonier@gmail.com">Daniel Stonier</maintainer>
+  <license>BSD</license>
+  <url type="website">http://ros.org/wiki/rocon_python_redis</url>
+  <url type="repository">https://github.com/robotics-in-concert/rocon_tools</url>
+  <url type="bugtracker">https://github.com/robotics-in-concert/rocon_tools/issues</url>
+  <author email="d.stonier@gmail.com">Daniel Stonier</author> -->
```
Comment (bit-pirate) [inline rocon_python_redis/package.xml]: remove email and trailing comment sign

Comment (bit-pirate) [inline rocon_python_redis/package.xml]: Maybe also add the author of the redis code?

Copyright (c) 2010 Andy McCurdy

File: rocon_python_redis/src/README.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
+# redis-py
+
+The Python interface to the Redis key-value store.
```
Comment (bit-pirate) [inline rocon_python_redis/src/README.md:3]: Shouldn't this readme be in the package root level and not in the src package?

File: rocon_python_redis/src/rocon_python_redis/client.py [hunk 5]
```diff
@@ -0,0 +1,1869 @@
...
+        except ConnectionError:
+            conn.disconnect()
+            # if we were watching a variable, the watch is no longer valid
+            # since this connection has died. raise a WatchError, which
+            # indicates the user should retry his transaction. If this is more
+            # than a temporary failure, the WATCH that the user next issue
+            # will fail, propegating the real ConnectionError
+            if self.watching:
+                raise WatchError("A ConnectionError occured on while watching "
+                                 "one or
```
Comment (bit-pirate) [inline rocon_python_redis/src/rocon_python_redis/client.py:1724]: Bloody long file!

File: rocon_python_utils/package.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  <version>0.1.0</version>
+  <description>
+    Python system and ros utilities.
+  </description>
+  <maintainer email="d.stonier@gmail.com">Daniel Stonier</maintainer>
+  <license>BSD</license>
+  <url type="website">http://ros.org/wiki/rocon_python_utils</url>
+  <url type="repository">https://github.com/robotics-in-concert/rocon_tools</url>
+  <url type="bugtracker">https://github.com/robotics-in-concert/rocon_tools/issues</url>
+  <author email="d.stonier@gmail.com">Daniel Stonier</author>
```
Comment (bit-pirate) [inline rocon_python_utils/package.xml]: remove email

File: rocon_interactions/src/rocon_interactions/exceptions.py (hunk 1 above)
Comment (stonier) [inline rocon_interactions/src/rocon_interactions/exceptions.py:8]: No, exceptions are built-in like int is in c++.

File: rocon_python_redis/src/README.md (hunk 4 above)
Comment (stonier) [inline rocon_python_redis/src/README.md:3]: This is part of the original distribution - everything under `src` has been intentionally left as intact as possible. If anything, the root level Readme.md could point here.

**Comment by bit-pirate** [comment]
Good to go.

**Pull request merged**